### PR TITLE
(SLV-184) Update bolt_helper to use '--run-as root' when running bolt commands

### PR DESCRIPTION
**Pull request by billclaytor**
* use '--run-as root' when running bolt commands
* update make_dir to use a single command

### CODE REVIEW

File: lib/ref_arch_setup/bolt_helper.rb [hunk 1]
```diff
@@ -39,7 +42,12 @@ class << self
     #
     # @return [true,false] Based on exit status of the bolt task
     def self.make_dir(dir, nodes)
-      cmd = "[ -d #{dir} ] || mkdir -p #{dir}"
+      # this causes an issue with bolt when 'run-as root' is not allowed
```
[the remark this answers is not in the document]
Comment (billclaytor) [reply lib/ref_arch_setup/bolt_helper.rb]: Updated

File: spec/ref_arch_setup/bolt_helper_spec.rb [hunk 2]
```diff
@@ -16,13 +16,52 @@
...
+      RefArchSetup::BoltHelper.bolt_options = bolt_user_opts
+      RefArchSetup::BoltHelper.init
+      expect(expect(RefArchSetup::BoltHelper.bolt_options_string)
+       .to(eq(default_bolt_options)))
+    end
+  end
+
+  describe "bolt_options" do
+    context "when the overwrite option is specified as true" do
+      it "uses on the user-specified options" do
```
Comment (RandellP) [inline spec/ref_arch_setup/bolt_helper_spec.rb]: typo?

File: lib/ref_arch_setup/bolt_helper.rb [hunk 3]
```diff
@@ -1,33 +1,67 @@
...
+      bolt_options(options_hash)
+    end
 
     # gets the bolt options as a string
     #
     # @author Sam Woods
     # @return [string] the string value for bolt options
     def self.bolt_options_string
       bolt_options_string = ""
+
```
Comment (RandellP) [inline lib/ref_arch_setup/bolt_helper.rb]: is this new newline intentional?

File: spec/ref_arch_setup/bolt_helper_spec.rb (hunk 2 above)
Comment (billclaytor) [reply spec/ref_arch_setup/bolt_helper_spec.rb]: Fixed

File: lib/ref_arch_setup/bolt_helper.rb (hunk 3 above)
Comment (billclaytor) [reply lib/ref_arch_setup/bolt_helper.rb]: Fixed

File: spec/ref_arch_setup/bolt_helper_spec.rb [hunk 4]
```diff
@@ -45,7 +44,7 @@
 
   describe "run_cmd_with_bolt" do
     before do
-      @expected_command = "bolt command run '#{cmd}' --nodes #{nodes}"
+      @expected_command = "bolt command run '#{cmd}' --nodes #{nodes} --run-as root"
```
[the remark this answers is not in the document]
Comment (billclaytor) [reply spec/ref_arch_setup/bolt_helper_spec.rb]: Updated

**Pull request merged**